### DISCUSSION TITLE
model piping: parameter name as a variable

### DISCUSSION THREAD

**Discussion by dzianismr** [nlmixrdevelopment/nlmixr#588] [Q&A] [answered]
Here is a question regarding model piping.  Lets take take an example from the model piping vignette:

A reference model:
fit <- nlmixr(one.compartment, theo_sd, est="focei",
control=list(print=0), table=list(npde=TRUE, cwres=TRUE))
Fixing a parameter:
one.ka.0.5 <- fit %>% ini(tka=fix(0.5)) %>%
nlmixr(est="focei", control=list(print=0), table=list(cwres=TRUE, npde=TRUE))

It may be more of an R question, but I hope someone can help me out with it.  What will be the right syntax  to provide model here parameter name as a variable to later loop (lapply, map etc) over a list of model parameters? 

A related discussion, however with the use of _update_, was posted here:
https://github.com/nlmixrdevelopment/nlmixr/issues/181

**Comment by mattfidler** [comment]
Hi @dzianismr 

Likely the most portable way to do this in a function is:

```r
f <- function(fit, par1, par2) {
eval(bquote(fit %>% ini(tka = .(par1), tcl = .(par2)))
}
```

I think it would be simply piping in the new `rxode2` but this works in the mean time.

**Reply by dzianismr** [reply]
Thanks @mattfidler. It is probably going in the right direction, however the main issue I have is with passing a parameter name e.g. 
parname = "tka"
f <- function(fit, parname, parval) {
eval(bquote(fit %>% ini(parname = .(parval)))
}
runs, however does not change tka value.

**Comment by mattfidler** [comment]
Well, this is a bit more complicated.

However all these things can be accessed in the underlying ui initial estimate data frame:

In `nlmixr` it is located at:
```r
fit$uif$ini
```

Because of techincal reasons you can assign it outside:

```r
ui <- fit$uif
# You can see what you need to modify by changing the ini to a data.frame
as.data.frame(ui$ini)
ini <- ui$ini

# Now by modifying the data frame you can fix/unfix parameters and modify an arbitrary parameter name.  Make sure the sorting remains itact

# Once the ini is modified you use the following:
assign("ini", ini, envir=ui)
```

Then the ui can be run with nlmixr

This code would be incompatible with the `rxode2` and `nlmixr2` interface since this section was rewritten.  However the basic structure is very similar, instead you would access `fit$ui$inifDf`

**Comment by dzianismr** [comment, marked as answer]
Thanks for the tips @mattfidler .

assign("ini", ini, envir=ui)
Does not work in my hands:
Error in assign("ini", ini, envir = ui) : invalid 'envir' argument

However given your tips I could built the following workaround: 
ui <- fit$uif
ii  <- which( ui$ini$name == parname )
ui$ini$est[ii] <- parval
ui$ini$fix[ii] <- TRUE
fit_new        <- nlmixr_fit( ui, theo_sd, est="focei" )

**Reply by mattfidler** [reply]
Perfect. This code should be easy to make comatible with nlmixr2
